root/etc/cont-init.d/50-config

```diff
@@ -41,32 +40,59 @@ if [ -n "${TEST_RUN}" ]; then
 fi
 
 # Create API key if needed
-if [ ! -f "/config/BOOKSTACK_APP_KEY.txt" ];
-    then
+if [ ! -f "/config/BOOKSTACK_APP_KEY.txt" ]; then
     echo "Generating BookStack app key for first run"
     key=$(php /app/www/artisan key:generate --show)
-    echo $key > /config/BOOKSTACK_APP_KEY.txt
-    echo "App Key set to $key you can modify the file to update /config/BOOKSTACK_APP_KEY.txt"
-elif [ -f "/config/BOOKSTACK_APP_KEY.txt" ];
-    then
+    echo "${key}" >/config/BOOKSTACK_APP_KEY.txt
+    echo "App Key set to ${key} you can modify the file to update /config/BOOKSTACK_APP_KEY.txt"
+elif [ -f "/config/BOOKSTACK_APP_KEY.txt" ]; then
     echo "App Key found - setting variable for seds"
     key=$(cat /config/BOOKSTACK_APP_KEY.txt)
 fi
 
 # .env file setup
 # check for the default app key or if it has been updated
-if grep -Fxq "APP_KEY=SomeRandomString" /config/www/.env || \
-! grep -Fxq "APP_KEY=${key}" /config/www/.env; then
+if ! grep -Fxq "APP_KEY=${key}" /config/www/.env; then
     sed -i "s#^APP_KEY=.*#APP_KEY=${key}#" /config/www/.env
 fi
-# check to see if db_user is set, if it is then run seds and if not then leave them
-if [ "${DB_USER}" ];
-	then
-	echo "Running config - db_user set"
-	sed -i "s/DB_HOST=.*/DB_HOST=${DB_HOST}/g" /config/www/.env
-	sed -i "s/DB_DATABASE=.*/DB_DATABASE=${DB_DATABASE}/g" /config/www/.env
-	sed -i "s/DB_USERNAME=.*/DB_USERNAME=${DB_USER}/g" /config/www/.env
-	sed -i "s/DB_PASSWORD=.*/DB_PASSWORD=${DB_PASS}/g" /config/www/.env
+
+# if DB_HOST contains a port and DB_HOST is not a IPv6 without brackets [..]
+# support ipv4:port, [ipv6]:port, and domain:port
+if [[ ${DB_HOST} =~ :[0-9]+$ ]] && ! [[ ${DB_HOST} =~ ^(:{0,2}[a-fA-F0-9]{1,4})+$ ]]; then
+    DB_HOST_PORT="${DB_HOST}"
+fi
+
+# if DB_HOST_PORT is set
+if [[ -n "${DB_HOST_PORT}" ]]; then
+    # if DB_PORT is set
+    if [[ -n "${DB_PORT}" ]]; then
+        echo "DB_PORT is not supported when using DB_HOST with port"
+        sleep infinity
+    fi
+    DB_HOST="${DB_HOST_PORT%:*}"
+    DB_PORT="${DB_HOST_PORT##*:}"
+fi
+
+# if DB_PORT is not set
+if [[ -z "${DB_PORT}" ]]; then
+    DB_PORT="3306"
+fi
+
+# check to see if DB_HOST is set, if it is then run seds and if not then leave them
+if [[ -n "${DB_HOST}" ]]; then
+    echo "Running config - DB_HOST set"
+
+    if ! grep -xq "^DB_PORT=.*" /config/www/.env; then
+        # add DB_PORT line to /config/www/.env because current /app/www/.env.example doesn't have it
+        sed -i "/^DB_HOST=.*/a DB_PORT=${DB_PORT}" /config/www/.env
+        echo "**** Insert DB_PORT=${DB_PORT} into /config/www/.env ****"
+    fi
+
+    sed -i "s/^DB_HOST=.*/DB_HOST=${DB_HOST}/g" /config/www/.env
```

Comment:
Thanks for the PR. While you're at it, can you change these seds to `"s/^[#]?DB_HOST=.*` so that even if upstream comments them out in the .env sample, they'll still be modified and enabled?

Reply:
Let me have a check

Reply:
I just updated the code to support `[#]?`, this case we need to use regex ERE to support zero or one, have to use `-E`
```
if ! grep -xqE "^[#]?DB_PORT=.*" /config/www/.env; then
...
sed -i -E "s/^[#]?DB_HOST=.*/DB_HOST=${DB_HOST}/g" /config/www/.env
```